examples/ros/src/smarts_ros/scripts/ros_driver.py

```diff
@@ -348,29 +345,80 @@ def _entity_to_vs(entity: EntityState) -> VehicleState:
         vs.speed = np.linalg.norm(vs.linear_velocity)
         return vs
 
+    class _VehicleStateVector:
+        def __init__(self, vs: VehicleState, stamp: float):
+            self.vs = vs
+            self.vector = np.concatenate(
+                (
+                    np.array((stamp,)),
+                    vs.pose.position,
+                    np.array((vs.pose.heading,)),
+                    np.array(vs.dimensions.as_lwh),
+                    vs.linear_velocity,
+                    vs.angular_velocity,
+                    vs.linear_acceleration,
+                    vs.angular_acceleration,
+                )
+            )
+
+        @property
+        def stamp(self):
+            return self.vector[0]
+
+        def average_with(self, other_vect: np.ndarray):
+            self.vector += other_vect
+            self.vector /= 2
+            self.update_vehicle_state()
```

Comment:
nit: From what I can see `vector`, `update_vehicle_state`, `average_with`, should be private members.

Reply:
The sub-class containing those three methods (`_VehicleStateVector`) is already marked as private, and these 3 things are being accessed by the "external" parent/owner class code (not just by methods in the subclass).  In such cases, I usually leave them public to denote that the owner class can access them too.